// react-native-scheduler/CalendarMonth/__stubsData/data.js
import moment from "moment";
import { Colours } from "../../shared/colour";
export var user1 = {
    id: "1",
    name: "Ben Af",
    colourIndicator: Colours.Cobalt,
    isSelected: false
};
export var user2 = {
    id: "2",
    name: "Cardano",
    colourIndicator: Colours.Cardinal,
    isSelected: false
};
export var user3 = {
    id: "3",
    name: "Zester",
    colourIndicator: Colours.Zest,
    isSelected: false
};
export var user4 = {
    id: "4",
    name: "Green Lantern",
    colourIndicator: Colours.LaPalma,
    isSelected: false
};
export var userCollections = [
    { id: "1", name: "Adam", colourIndicator: "blue" },
    { id: "2", name: "Adi", colourIndicator: "green" },
    { id: "3", name: "Bruno", colourIndicator: "red" },
    { id: "4", name: "George", colourIndicator: "pink" },
    { id: "5", name: "Chiro", colourIndicator: "black" },
    { id: "6", name: "Olivia", colourIndicator: "grey" }
];
var date1 = moment().add(1, "days");
var date2 = moment().add(2, "days");
var date3 = moment().add(3, "days");
var date4 = moment().add(4, "days");
var date5 = moment().add(1, "weeks");
var date6 = moment().add(1, "weeks").subtract(1, "days");
var date10 = moment().add(1, "month");
var date11 = moment().add(1, "month").add(1, "days");
var date12 = moment().add(1, "month").add(2, "days");
export var taskAndUserCollections = [
    {
        user: user1,
        tasks: [
            {
                id: "1-1",
                name: "Clean Toilet",
                startTime: date1.toDate(),
                endTime: date1.add(2, "hours").toDate()
            },
            {
                id: "1-2",
                name: "Gardening",
                startTime: date2.add(1, "hours").toDate(),
                endTime: date2.add(3, "hours").toDate()
            },
            {
                id: "1-3",
                name: "Gardening",
                startTime: date6.toDate(),
                endTime: date6.add(4, "hours").toDate()
            }
        ]
    },
    {
        user: user2,
        tasks: [
            {
                id: "2-1",
                name: "Clean Toilet",
                startTime: date1.add(3, "hours").toDate(),
                endTime: date1.add(5, "hours").toDate()
            },
            {
                id: "2-2",
                name: "Gardening",
                startTime: date2.add(2, "hours").toDate(),
                endTime: date2.add(4, "hours").toDate()
            }
        ]
    }
];
export var taskSchedules = {
    1: [
        {
            id: "1-1",
            name: "Clean Toilet",
            startTime: date1.toDate(),
            endTime: date1.add(2, "hours").toDate()
        },
        {
            id: "1-2",
            name: "Gardening",
            startTime: date2.add(1, "hours").toDate(),
            endTime: date2.add(3, "hours").toDate()
        },
        {
            id: "1-3",
            name: "Gardening",
            startTime: date6.toDate(),
            endTime: date6.add(4, "hours").toDate()
        }
    ],
    2: [
        {
            id: "2-1",
            name: "Clean Toilet",
            startTime: date1.add(3, "hours").toDate(),
            endTime: date1.add(5, "hours").toDate()
        },
        {
            id: "2-2",
            name: "Gardening",
            startTime: date2.add(2, "hours").toDate(),
            endTime: date2.add(4, "hours").toDate()
        }
    ],
    3: [
        {
            id: "3-1",
            name: "Clean Toilet",
            startTime: date3.add(1, "hours").toDate(),
            endTime: date3.add(5, "hours").toDate()
        },
        {
            id: "3-2",
            name: "Gardening",
            startTime: date4.add(2, "hours").toDate(),
            endTime: date4.add(4, "hours").toDate()
        }
    ],
    4: [
        {
            id: "4-1",
            name: "BBQ",
            startTime: date1.add(4, "hours").toDate(),
            endTime: date1.add(6, "hours").toDate()
        },
        {
            id: "4-2",
            name: "Clean Toilet",
            startTime: date5.add(1, "hours").toDate(),
            endTime: date5.add(5, "hours").toDate()
        },
        {
            id: "4-3",
            name: "Gardening",
            startTime: date6.toDate(),
            endTime: date6.add(4, "hours").toDate()
        }
    ]
};
